### PR TITLE
rail: added verbose logs

### PR DESCRIPTION
**Pull request by hardening**
People are working on RAIL these days, these extra logs helps a lot.
The PR also contains some reformatting (astyle has been unfortunate for some lines).

### CODE REVIEW

File: libfreerdp/core/window.c [hunk 1]
```diff
@@ -498,6 +497,148 @@ static BOOL window_order_supported(const rdpSettings* settings, UINT32 fieldFlag
...
+#define DUMP_APPEND(...) do { \
+		snprintf(tmpBuffer, sizeof(tmpBuffer), __VA_ARGS__); \
+		strncat(buffer, tmpBuffer, rem); \
+		rem -= strlen(tmpBuffer); \
+		} while(0)
+
+
+static void dump_window_state_order(wLog *log, const char *msg, const WINDOW_ORDER_INFO* order, const WINDOW_STATE_ORDER *state)
+{
+	char buffer[3000];
```
Comment (akallabeth) [inline libfreerdp/core/window.c]: a `= { 0 };` might be a good idea

Comment (hardening) [reply libfreerdp/core/window.c]: Not needed, the first call is a snprintf, that already adds the  ̀\0`

**Comment by akallabeth** [comment]
looks good overall, formatting is a bit off in the new functions ;)
(single line cases and such stuff)

**Comment by hardening** [comment]
@freerdp-bot can you do a correct test please, even on the windows host ?

**Comment by hardening** [comment]
@akallabeth about formatting we have such form in other places in the code. But I can change it if it's a show stopper.

**Comment by akallabeth** [comment]
@hardening shouldn´t be a big issue. If we finally can merge #4350 we'll do a reformatting anyway ;)

**Comment by hardening** [comment]
@akallabeth ok so feel free to review and merge ;)

**Comment by hardening** [comment]
merge ?

**Pull request merged**